resources/wikia/modules/spec/geo.spec.js

```diff
@@ -29,46 +29,60 @@ describe("Geo", function () {
 		expect(typeof geo.getContinentCode).toBe('function');
 	});
 
-	it('getGeoData returns parsed data', function() {
+	it('getGeoData returns parsed data', function () {
 		var geoData = geo.getGeoData();
 
 		expect(typeof geoData).toBe('object');
 		expect(geoData.city).toBe('Poznan');
 	});
 
-	it('returns country and continent code', function() {
+	it('returns country and continent code', function () {
 		expect(geo.getCountryCode()).toBe('PL');
 		expect(geo.getContinentCode()).toBe('EU');
 	});
 
-	it('returns region code', function() {
+	it('returns region code', function () {
 		expect(geo.getRegionCode()).toBe('72');
 	});
 
-	it('isProperCountry test', function() {
-		expect(geo.isProperCountry(['ZZ', 'PL'])).toBeTruthy();
+	it('isProperCountry test', function () {
+		expect(geo.isProperCountry(['ZZ', 'PL', 'non-YY'])).toBeTruthy();
 		expect(geo.isProperCountry(['ZZ', 'YY'])).toBeFalsy();
+		expect(geo.isProperCountry(['ZZ', 'non-PL'])).toBeFalsy();
+		expect(geo.isProperCountry(['ZZ', 'PL', 'non-PL'])).toBeFalsy();
 	});
 
-	it('isProperRegion test', function() {
-		expect(geo.isProperRegion(['ZZ', 'PL-72'])).toBeTruthy();
+	it('isProperRegion test', function () {
+		expect(geo.isProperRegion(['ZZ', 'PL-72', 'non-PL-33'])).toBeTruthy();
 		expect(geo.isProperRegion(['ZZ', 'PL-33'])).toBeFalsy();
+		expect(geo.isProperRegion(['ZZ', 'non-PL-72'])).toBeFalsy();
+		expect(geo.isProperRegion(['ZZ', 'PL-72', 'non-PL-72'])).toBeFalsy();
 	});
 
-	it('isProperContinent test', function() {
-		expect(geo.isProperContinent(['XX-NA','XX-EU'])).toBeTruthy();
+	it('isProperContinent test', function () {
+		expect(geo.isProperContinent(['XX-NA','XX-EU', 'non-XX-SA'])).toBeTruthy();
 		expect(geo.isProperContinent(['XX-NA','XX-SA'])).toBeFalsy();
+		expect(geo.isProperContinent(['XX-NA','XX-SA', 'non-XX-EU'])).toBeFalsy();
+		expect(geo.isProperContinent(['XX-NA','XX-EU', 'non-XX-EU'])).toBeFalsy();
 	});
 
-	it('isProperContinent test - any region (XX)', function() {
+	it('isProperContinent test - any region (XX)', function () {
 		expect(geo.isProperContinent(['XX'])).toBeTruthy();
+		expect(geo.isProperContinent(['XX', 'non-XX-SA'])).toBeTruthy();
+		expect(geo.isProperContinent(['XX', 'non-XX-EU'])).toBeFalsy();
 	});
 
-	it('isProperGeo test', function() {
+	it('isProperGeo test', function () {
 		expect(geo.isProperGeo(['PL'])).toBeTruthy();
 		expect(geo.isProperGeo(['XX-EU'])).toBeTruthy();
 		expect(geo.isProperGeo(['XX'])).toBeTruthy();
 		expect(geo.isProperGeo(['PL-72'])).toBeTruthy();
+		expect(geo.isProperGeo(['non-PL'])).toBeFalsy();
```

Comment:
As discussed offline: let's make only `isProperGeo` public and move important tests from above here.